custodia/custodia

```diff
@@ -44,16 +47,24 @@ def attach_store(typename, plugins, stores):
 CONFIG_SPECIALS = ['authenticators', 'authorizers', 'consumers', 'stores']
 
 def parse_config(cfgfile):
-    parser = RawConfigParser()
+    parser = SafeConfigParser(interpolation=ExtendedInterpolation())
     parser.optionxform = str
-    files = parser.read(cfgfile)
-    if len(files) == 0:
-        raise IOError("Failed to read config file")
+
+    with open(cfgfile) as f:
+        parser.readfp(f)
 
     config = dict()
     for s in CONFIG_SPECIALS:
         config[s] = dict()
+
+    # add env
+    parser['ENV'] = {
+        k: v.replace('$', '$$') for k, v in os.environ.items()
+        if not set(v).intersection('\r\n\x00')}
+
     for s in parser.sections():
+        if s == 'ENV':
```

Comment:
It's a new section that contains all variables from os.environ. The section is populated a couple of lines earlier. I have to ignore the section here. Otherwise  the function runs into raise ValueError('Invalid section name [%s].\n' % s).

The new new interpolation is explained https://docs.python.org/3.5/library/configparser.html#configparser.ExtendedInterpolation I can add a couple of examples.